### PR TITLE
Add Shared Libraries team in codeowners file

### PR DESCRIPTION
**Pull request by desi**
Based on Github Teams Proposal we want to update codeowners to reflect the team(s) that should be consulted before a file is changed. In this case we are adding the Shared Libraries team.

### CODE REVIEW

**Review by MajorLift — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # Lines starting with '#' are comments.
 # Each line is a file pattern followed by one or more owners.
 
-*                                    @MetaMask/devs
+*                                   @MetaMask/shared-libraries-engineers
```
Comment (Mrtenz) [inline .github/CODEOWNERS:4]:
```suggestion
*                                   @MetaMask/shared-libraries-engineers

```

Comment (desi) [reply .github/CODEOWNERS]: Fixed in [6d4d99c](https://github.com/MetaMask/eslint-config/pull/343/commits/6d4d99cb353485899dc9b4b7b5c8dc571ac81b9d)

Comment (mcmire) [inline .github/CODEOWNERS:4]: Following standup this morning, maybe we want to add:

``` suggestion
*                                   @MetaMask/shared-libraries-engineers @Mrtenz 
```

or even:

``` suggestion
*                                   @MetaMask/shared-libraries-engineers @MetaMask/snaps-devs 
```

?

Comment (Mrtenz) [reply .github/CODEOWNERS:4]: I think adding Snaps Devs is fine, and we can add other teams if they want to?

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**